### PR TITLE
Ensure leaders are evenly distributed before proceeding with asymmetric partition test

### LINKED ISSUE #9266: AsymmetricNetworkPartitionIT.shouldWithstandAsymmetricNetworkPartition
**Summary**

- How often does the test fail? at least twice
- Does it block your work? no
- Do we suspect that it is a real failure? I don't think so

**Failures**

<details><summary>Example assertion failure</summary>
<pre>
java.lang.AssertionError: 

Expecting actual:
  0
not to be equal to:
  0

	at io.camunda.zeebe.it.clustering.network.AsymmetricNetworkPartitionIT.shouldWithstandAsymmetricNetworkPartition(AsymmetricNetworkPartitionIT.java:77)
</pre>
</details>

**Hypotheses**

None so far

**Logs**

None available. Failing build is: https://github.com/camunda/zeebe/actions/runs/2256830107

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR resolves the flakiness in `AsymmetricNetworkPartitionIT` by ensuring that we have different leaders for all partitions before proceeding with the test case. It does so by using the rebalancing API in combination with priority election.

I took the opportunity to also clean up the test a bit. The first commit is the fix itself. The second commit cleans up a little by making use of new APIs in `ZeebeCluster` and reusing the Testcontainers JUnit 5 extension. The final commit makes the cluster static, such that it's reused by all test cases. Utilities are only installed once at the beginning, and we guarantee unreachable routes are flushed between cases. I also reduced the number of partitions 2 if only to make the containers a little lighter - I didn't see a big need for 3 partitions, to be honest.

I'm happy to split this and take the last two commits in a separate PR if you'd prefer.

## Related issues

closes #9266 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by npepinpe** [comment]
I also realized you could probably write this test using Toxiproxy, but I think it would be a bit more complicated to keep it controlled. You could have a cluster of 3 nodes, one node is not proxied and also talks to the other two without a proxy, and the other two talk to each other via a proxy. To cut off communication between these is super easy, you just block one of the proxies via `ContainerProxy` (there's a method for it). Reverting is just as easy.

The problem is that you need to know the proxy addresses before starting the cluster, which means you kind of need to know which node is leader - or to be able to enforce which node will be leader of which partition. Maybe using fixed partitioning would work, but then I'm not sure if it's worth the more complex setup :man_shrugging:

**Comment by npepinpe** [comment]
Not sure how this would mess up the exporter tests, but considering I'm in the middle of refactoring them, let's ignore them here.

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/network/AsymmetricNetworkPartitionIT.java [hunk 1]
```diff
@@ -64,18 +73,22 @@ void afterEach() {
...
       final AsymmetricNetworkPartitionTestCase asymmetricNetworkPartitionTestCase)
       throws IOException, InterruptedException {
     // given
     setupZeebeCluster();
     zeebeClient = cluster.newClientBuilder().build();
 
+    // the test only works if the leaders of partition 1 and 3 are different nodes
+    Awaitility.await("partition 1 and 3 have a different leader")
+        .atMost(Duration.ofSeconds(30))
+        .during(Duration.ofSeconds(5))
```
Comment (Zelldon) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/network/AsymmetricNetworkPartitionIT.java:116]: :+1: Interesting.

**Review by ChrisKujawa — APPROVED** [review]
Thanks good job:+1: :medal_military:

**Comment by npepinpe** [comment]
bors merge

**Comment by npepinpe** [comment]
Oh no, this is a new flaky test that I wrote last week :see_no_evil:

**Pull request merged**